### PR TITLE
Protocol v11 XDR changes

### PR DESCRIPTION
**Pull request by bartekn**
This PR updates XDR files to be up to date with stellar-core v11. Changes affecting this repository:
* Operation type names changes:
  * `manage_offer` -> `manage_sell_offer`,
  * `create_passive_offer` -> `create_passive_sell_offer`,
  * New `manage_buy_offer` operation.
* Offer IDs are now of `int64` type.

For backward compatibility, `manage_offer` and `create_passive_offer` type names have not been changed in Horizon. This will be a breaking change in Horizon 0.18.0.

Builders for `manage_buy_offer` operation has **not** been added in this PR (new issue: https://github.com/stellar/go/issues/1165).

### CODE REVIEW

File: services/horizon/internal/ingest/session.go [hunk 1]
```diff
@@ -748,8 +756,19 @@ func (is *Session) operationDetails() map[string]interface{} {
...
+		op := c.Operation().Body.MustManageBuyOfferOp()
+		details["offer_id"] = op.OfferId
+		details["amount"] = amount.String(op.BuyAmount)
+		details["price"] = op.Price.String()
+		details["price_r"] = map[string]interface{}{
+			"n": op.Price.N,
+			"d": op.Price.D,
+		}
+		is.assetDetails(details, op.Buying, "buying_")
+		is.assetDetails(details, op.Selling, "selling_")
```
Comment (bartekn) [inline services/horizon/internal/ingest/session.go]: Note to self: confirm this should/shouldn't be reversed.

**Comment by theaeolianmachine** [comment]
@poliha it is! It's specified in [CAP-0006](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0006.md).

**Comment by theaeolianmachine** [comment]
@bartekn this might be useful for you: https://github.com/stellar/stellar-core/pull/2039

File: exp/txnbuild/manage_offer.go [hunk 2]
```diff
@@ -7,10 +7,10 @@ import (
 	"github.com/stellar/go/xdr"
 )
 
-//CreateOfferOp returns a ManageOffer operation to create a new offer, by
+//CreateOfferOp returns a ManageSellOffer operation to create a new offer, by
```
Comment (ire-and-curses) [inline exp/txnbuild/manage_offer.go]: What about ManageBuyOffer? Are you leaving adding the new operation to `txnbuild` out of this PR?

File: services/horizon/internal/docs/reference/resources/operation.md [hunk 3]
```diff
@@ -221,10 +222,10 @@ A path payment operation represents a payment from one account to another throug
 }
 ```
 
-<a id="manage-offer"></a>
-### Manage Offer
+<a id="manage-sell-offer"></a>
+### Manage Sell Offer
 
-A "Manage Offer" operation can create, update or delete an
+A "Manage Offer" operation can create, update or delete a sell
```
Comment (ire-and-curses) [inline services/horizon/internal/docs/reference/resources/operation.md]: A "Manage Offer" -> A "Manage Sell Offer"

File: services/horizon/internal/docs/reference/resources/operation.md [hunk 4]
```diff
@@ -290,12 +291,12 @@ offers or payments, this offer can potentially be filled.
...
+  "type": "manage_offer" // `manage_sell_offer` from v0.18.0
 }
 ```
 
-<a id="create-passive-offer"></a>
-### Create Passive Offer
+<a id="create-passive-sell-offer"></a>
+### Create Passive Sell Offer
 
 “Create Passive Offer” operation creates an offer that won't consume a counter offer that exactly matches this offer. This is useful for offers just used as 1:1 exchanges for path payments. Use Manage Offer to manage this offer after using this operation to create it.
```
Comment (ire-and-curses) [inline services/horizon/internal/docs/reference/resources/operation.md]: "Create Passive Offer" operation -> "Create Passive Sell Offer" operation

File: xdr/Stellar-ledger-entries.x [hunk 5]
```diff
@@ -283,6 +283,7 @@ enum EnvelopeType
 {
     ENVELOPE_TYPE_SCP = 1,
     ENVELOPE_TYPE_TX = 2,
-    ENVELOPE_TYPE_AUTH = 3
+    ENVELOPE_TYPE_AUTH = 3,
+    ENVELOPE_TYPE_SCPVALUE = 4
```
Comment (ire-and-curses) [inline xdr/Stellar-ledger-entries.x]: Is this new envelope type documented anywhere?

File: xdr/Stellar-ledger.x [hunk 6]
```diff
@@ -10,12 +10,24 @@ namespace stellar
 
 typedef opaque UpgradeType<128>;
 
+enum StellarValueType
```
Comment (ire-and-curses) [inline xdr/Stellar-ledger.x]: Is it worth a comment here to explain what this is?

File: xdr/Stellar-transaction.x [hunk 7]
```diff
@@ -468,29 +490,29 @@ default:
...
+    MANAGE_SELL_OFFER_BUY_NOT_AUTHORIZED = -5,  // not authorized to buy
+    MANAGE_SELL_OFFER_LINE_FULL = -6,      // can't receive more of what it's buying
+    MANAGE_SELL_OFFER_UNDERFUNDED = -7,    // doesn't hold what it's trying to sell
+    MANAGE_SELL_OFFER_CROSS_SELF = -8,     // would cross an offer from the same user
+    MANAGE_SELL_OFFER_SELL_NO_ISSUER = -9, // no issuer for what we're selling
+    MANAGE_SELL_OFFER_BUY_NO_ISSUER = -10, // no issuer for what we're buying
 
     // update errors
-    MANAGE_OFFER_NOT_FOUND = -11, // offerID does not match an existing offer
+    MANAGE_SELL_OFFER_NOT_FOUND = -11, // offerID does not match an existing offer
```
Comment (ire-and-curses) [inline xdr/Stellar-transaction.x]: extra blank line

File: xdr/Stellar-transaction.x [hunk 8]
```diff
@@ -516,9 +538,42 @@ struct ManageOfferSuccessResult
...
+    MANAGE_BUY_OFFER_BUY_NOT_AUTHORIZED = -5,  // not authorized to buy
+    MANAGE_BUY_OFFER_LINE_FULL = -6,      // can't receive more of what it's buying
+    MANAGE_BUY_OFFER_UNDERFUNDED = -7,    // doesn't hold what it's trying to sell
+    MANAGE_BUY_OFFER_CROSS_SELF = -8,     // would cross an offer from the same user
+    MANAGE_BUY_OFFER_SELL_NO_ISSUER = -9, // no issuer for what we're selling
+    MANAGE_BUY_OFFER_BUY_NO_ISSUER = -10, // no issuer for what we're buying
+
+    // update errors
+    MANAGE_BUY_OFFER_NOT_FOUND = -11, // offerID does not match an existing offer
+
```
Comment (ire-and-curses) [inline xdr/Stellar-transaction.x]: extra blank line

File: xdr/Stellar-transaction.x [hunk 9]
```diff
@@ -693,7 +748,9 @@ enum OperationResultCode
 
     opBAD_AUTH = -1,     // too few valid signatures / wrong network
     opNO_ACCOUNT = -2,   // source account was not found
-    opNOT_SUPPORTED = -3 // operation not supported at this time
+    opNOT_SUPPORTED = -3, // operation not supported at this time
+    opTOO_MANY_SUBENTRIES = -4, // max number of subentries already reached
```
Comment (ire-and-curses) [inline xdr/Stellar-transaction.x]: Are these new errors documented anywhere?

File: exp/txnbuild/manage_offer.go (hunk 2 above)
Comment (bartekn) [reply exp/txnbuild/manage_offer.go]: Yes: https://github.com/stellar/go/issues/1165 I just updated the current code to be able to compile and pass tests.

File: xdr/Stellar-ledger-entries.x (hunk 5 above)
Comment (bartekn) [reply xdr/Stellar-ledger-entries.x]: These is copied directly from https://github.com/stellar/stellar-core/tree/master/src/xdr We shouldn't make any changes to these files (including comments).

File: xdr/Stellar-ledger.x (hunk 6 above)
Comment (bartekn) [reply xdr/Stellar-ledger.x]: These is copied directly from https://github.com/stellar/stellar-core/tree/master/src/xdr We shouldn't make any changes to these files (including comments).

File: xdr/Stellar-transaction.x (hunk 7 above)
Comment (bartekn) [reply xdr/Stellar-transaction.x]: These is copied directly from https://github.com/stellar/stellar-core/tree/master/src/xdr We shouldn't make any changes to these files (including comments).

File: xdr/Stellar-transaction.x (hunk 8 above)
Comment (bartekn) [reply xdr/Stellar-transaction.x]: These is copied directly from https://github.com/stellar/stellar-core/tree/master/src/xdr We shouldn't make any changes to these files (including comments).

File: xdr/Stellar-ledger-entries.x (hunk 5 above)
Comment (ire-and-curses) [reply xdr/Stellar-ledger-entries.x]: ok - shall we open a bug report for core then?

Comment (bartekn) [reply xdr/Stellar-ledger-entries.x]: I think so, yes.

File: services/horizon/internal/ingest/session.go (hunk 1 above)
Comment (bartekn) [reply services/horizon/internal/ingest/session.go]: After reading [CAP-0006](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0006.md) I think we shouldn't invert price to properly display operations submitted in a transaction. In stellar-core price is inverted when offer was not executed entirely and must be added to the ledger.

**Pull request merged**